### PR TITLE
Code Cleanup: Remove Unused Imports, Update UI/Auth, Memoize Markdown

### PR DESCRIPTION
**Pull request by AndlerRL**
@coderabbitai provide a description using changelog format.

## Summary by Sourcery

Refactor and improve various components and utilities across the masterbots.ai application, focusing on code organization, performance, and user experience

Enhancements:
- Introduced a centralized markdown components utility for consistent rendering across different components
- Added a device detection hook to support responsive design
- Improved code block and markdown rendering with more flexible processing
- Simplified routing and navigation logic in various components

Tests:
- Ensured consistent markdown rendering across different components
- Improved error handling and edge cases in utility functions

Chores:
- Updated Next.js and React dependencies to latest versions
- Cleaned up unused imports and simplified component logic
- Improved type safety and reduced code duplication

## Summary by CodeRabbit

- **New Features**
  - Added a back navigation element on thread pages to facilitate easier user movement.
  - Updated authentication flows to redirect users to a new landing path after sign in/up.

- **Refactor**
  - Consolidated Markdown rendering logic across messaging components for streamlined processing.
  - Simplified navigation parameter handling and refined message sorting for clearer conversation flow.

- **Style**
  - Adjusted responsive layouts and refined button texts for a cleaner, more intuitive interface.
  - Enhanced code block support with expanded programming language options.

- **Chores**
  - Upgraded key dependencies for improved stability and performance.

### CODE REVIEW

**Comment by AndlerRL** [comment]
@coderabbitai review

File: apps/masterbots.ai/lib/memoized-markdown-components.tsx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+import type { Components } from 'react-markdown'
+
+/**
+ * Preprocesses the children to combine adjacent nodes with a colon.
+ *
+ * @param children The children nodes.
+ * @returns The preprocessed children.
+ */
+
+const preprocessChildren = (children: React.ReactNode): React.ReactNode => {
```
[the remark this answers is not in the document]
Comment (AndlerRL) [reply apps/masterbots.ai/lib/memoized-markdown-components.tsx:14]: @coderabbitai please proceed to track this observation within a new issue.

File: apps/masterbots.ai/lib/hooks/use-mb-chat.tsx [hunk 2]
```diff
@@ -193,16 +193,38 @@ export function MBChatProvider({ children }: { children: React.ReactNode }) {
...
+
+		return {
+			checks: {
+				isContinuingThread: Boolean(continuousThreadId),
+			},
+			params: {
+				continuousThreadId: continuousThreadId,
+			},
+		}
+	}
```
[the remark this answers is not in the document]
Comment (AndlerRL) [reply apps/masterbots.ai/lib/hooks/use-mb-chat.tsx:216]: @coderabbitai it is client-side all the time. This component needs to be in the client-side to be able to work, the reason why the "use-client" directive is included.

File: apps/masterbots.ai/components/routes/chat/chat-panel.tsx [hunk 3]
```diff
@@ -124,7 +125,7 @@ export function ChatPanel({
 									isActive={webSearch}
 									onChange={(newValue) => {
 										console.log('ChatPanel: Toggle Web Search:', newValue)
-										setWebSearch(newValue)
+										setWebSearch()
```
[the remark this answers is not in the document]
Comment (AndlerRL) [reply apps/masterbots.ai/components/routes/chat/chat-panel.tsx:128]: @coderabbitai I know this, but we will handle this later. Please proceed to track this into a new issue and assign @Bran18 to it.

**Comment by AndlerRL** [comment]
[Andler Development Support - March 2025](https://trello.com/c/gEWUoW41/319-andler-development-support-march-2025)
